File: Practice/283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        //striver a2z array-easy
        int count=0;
        int n = nums.size();

        for(int i=0; i<n; i++){
            if(nums[i]==0)
                count++;
            else
                nums[i-count]=nums[i];
        }

        while(count){
            nums[n-count]=0;
            count--;
        }
    }
};

File: Practice/283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        count = 0
        n = len(nums)

        for i in range(n):
            if nums[i]==0:
                count += 1
            else:
                nums[i-count] = nums[i]

        while count>0:
            nums[n-count] = 0
            count -=1
        